src/template.rs

```diff
@@ -95,30 +96,61 @@ impl FunctionTemplate {
             value::Function::from_raw(&self.0, raw)
         }
     }
+
+    /// Returns the underlying raw pointer behind this function template.
+    pub fn as_raw(&self) -> v8::FunctionTemplateRef {
+        self.1
+    }
 }
 
 impl ObjectTemplate {
     /// Creates an ObjectTemplate.
     pub fn new(isolate: &isolate::Isolate) -> ObjectTemplate {
+        ObjectTemplate::<()>::with_internal(isolate)
+    }
+
+    /// Creates an ObjectTemplate<T> where T is the type of the internal rust object.
+    pub fn with_internal<T>(isolate: &isolate::Isolate) -> ObjectTemplate<T> {
```

Comment:
The use case I have is to back V8 objects with rust objects so that the V8 object can callback into rust code and access that object. Basically I'd like to port this example from the V8 embedder's guide to rust: https://github.com/v8/v8/wiki/Embedder's%20Guide#accessing-dynamic-variables